cmd/format-erasure.go

```diff
@@ -307,8 +307,23 @@ func countErrs(errs []error, err error) int {
 }
 
 // Check if unformatted disks are equal to write quorum.
-func quorumUnformattedDisks(errs []error) bool {
-	return countErrs(errs, errUnformattedDisk) >= (len(errs)/2)+1
+func quorumUnformattedDisks(errs []error, drives []StorageAPI) bool {
+	setErrs := map[int][]error{}
+	for i, drive := range drives {
+		serrs, ok := setErrs[drive.Endpoint().SetIdx]
+		if ok {
+			serrs = append(serrs, errs[i])
+		} else {
+			serrs = []error{errs[i]}
+		}
+		setErrs[drive.Endpoint().SetIdx] = serrs
+	}
+
+	ok := true
+	for _, serrs := range setErrs {
+		ok = ok && (countErrs(serrs, nil) > len(serrs)/2)
+	}
+	return !ok
```

Comment:
we need to look for online disks across sets, instead of just blindly looking at all errors/2 + 1